### PR TITLE
Replace the rest of Gradle scripts with gradle.kts, extract test utilities to a separate module

### PR DESCRIPTION
**Pull request by dkhalanskyjb**
I think the best way to review this is commit-by-commit.

### CODE REVIEW

**Comment by qwwdfsad** [comment]
A very promising branch name 😄

**Review by qwwdfsad — APPROVED** [review]
Superb!

File: reactive/kotlinx-coroutines-reactor/build.gradle.kts [hunk 1]
```diff
@@ -23,12 +23,10 @@ java {
 
 tasks {
     compileKotlin {
-        kotlinOptions.jvmTarget = "1.8"
```
[the remark this answers is not in the document]
Comment (dkhalanskyjb) [reply reactive/kotlinx-coroutines-reactor/build.gradle.kts:26]: In `develop`, `buildSrc/src/main/kotlin/kotlin-jvm-conventions.gradle.kts` contains

```kotlin
java {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8
}

kotlin {
    compilerOptions {
        jvmTarget = JvmTarget.JVM_1_8
    }
    jvmToolchain(jdkToolchainVersion)
}
```

I think it's a safe guess that these are not required for some time already. I can check more thoroughly if you have further doubts.

File: buildSrc/src/main/kotlin/configure-compilation-conventions.gradle.kts [hunk 2]
```diff
@@ -33,8 +33,24 @@ configure(subprojects) {
                     "-Xno-receiver-assertions"
                 )
             }
+            if (this@configureEach is KotlinNativeCompile) {
+                optIn.addAll(
+                    "kotlinx.cinterop.ExperimentalForeignApi",
+                    "kotlinx.cinterop.UnsafeNumber",
+                    "kotlin.experimental.ExperimentalNativeApi",
```
[the remark this answers is not in the document]
Comment (dkhalanskyjb) [reply buildSrc/src/main/kotlin/configure-compilation-conventions.gradle.kts:40]: Got it. For the record, this was already in `core/build.gradle`:
```
kotlin.sourceSets.configureEach {
    // Do not apply 'ExperimentalForeignApi' where we have allWarningsAsErrors set
    if (it.name in ["jvmMain", "jvmCoreMain", "jsMain", 'wasmJsMain', 'jsAndWasmSharedMain', "concurrentMain", "commonMain"]) return
    languageSettings {
        optIn('kotlinx.cinterop.ExperimentalForeignApi')
        optIn('kotlin.experimental.ExperimentalNativeApi')
    }
}
```
My addition was only `UnsafeNumber`.

[the remark this answers is not in the document]
Comment (qwwdfsad) [reply buildSrc/src/main/kotlin/configure-compilation-conventions.gradle.kts:40]: This one is a leftover for train builds as well. Proceed as you see fit here

**Comment by dkhalanskyjb** [comment]
Status: checked the classfile versions, POM files, module-info, version files in jars, and consumption from an Android project, JVM project, and a multiplatform project, including IDE navigation. The Ktor build is in progress, but so far nothing's failed. Everything seems to work.
@qwwdfsad, do you have preferences w.r.t how this should be squashed?

**Comment by qwwdfsad** [comment]
I'm okay with squashing it all into a single commit, no hard preferences here

**Pull request merged**